src/test/java/com/github/sdonkov/PrefixTreeTest.java

```diff
@@ -0,0 +1,73 @@
+package com.github.sdonkov;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+public class PrefixTreeTest {
+
+    private static List<CharSequence> words;
+    private static PrefixTree prefixTree;
+
+    @BeforeAll
+    static void setUp() throws IOException, URISyntaxException {
+        words = new ArrayList<>();
+        prefixTree = new PrefixTree();
+        List<String> lines = Files.readAllLines(Paths.get(PrefixTreeTest.class.getResource("/tests.txt").toURI()));
+        words.addAll(lines);
```

Comment:
Is this because `words` is not `List<String>`?
You can simply make it so, nothing will break as far as I can tell and you'll avoid declaring `lines`. 
If you are curious on how `addAll` here works see Effective Java's "Item 31: Use bounded wildcards to increase API flexibility" - it's among the challenging parts of the book.